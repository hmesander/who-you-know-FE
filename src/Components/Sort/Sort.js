import React, { Component } from 'react';
import Card from '../Card/Card.js';
import mockData from '../../helpers/mockData';

export default class Sort extends Component {
  constructor() {
    super();

    this.state = {
      cards: []
    };
  }

  async componentDidMount() {
    const cards = await this.fetchCardsToSort();
    this.setState({
      cards
    });
  }

  fetchCardsToSort = () => {
    console.log('fetch sort');
    return mockData;
  }

  getNewCard = () => {
    const newCards = [...this.state.cards];
    let cardToSave = newCards.shift();
    this.setState({
      cards: newCards
    });
    return cardToSave;
  }

  handleUpVote = () => {
    let cardToSave = this.getNewCard();
    console.log(cardToSave);
  }

  handleMiddleVote = () => {
    let cardToSave = this.getNewCard();
  }

  handleDownVote = () => {
    let cardToSave = this.getNewCard();
  }


  render() {
    return (
      <div>
        <Card cards={this.state.cards} handleUpVote={this.handleUpVote} handleMiddleVote={this.handleMiddleVote} handleDownVote={this.handleDownVote}/>
      </div>
    );
  }
}