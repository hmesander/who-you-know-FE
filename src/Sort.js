import React, { Component } from 'react';
import Card from './Card.js';

export default class Sort extends Component {
  constructor() {
    super();

  }

  render() {
    return (
      <div>
        <Card />
      </div>
    );
  }
}