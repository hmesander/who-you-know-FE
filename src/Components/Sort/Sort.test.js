/*eslint-disable no-undef*/

import React from 'react';
import Sort from './Sort';
import { shallow } from 'enzyme';


describe('SORT TESTS', () => {
  it('should match the snapshot if there is no user id', () => {
    const wrapper = shallow(<Sort />, { disableLifecycleMethods: true });

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if there is a user id', () => {
    const wrapper = shallow(<Sort userId={1} />, { disableLifecycleMethods: true });

    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should call update difficulty level with a argument of hard when handleVote is invoked', () => {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve());
    const wrapper = shallow(<Sort userId={2} />, { disableLifecycleMethods: true });

    const spy = spyOn(wrapper.instance(), 'updateDifficultyLevel');
    wrapper.instance().setState({
      cards: [{id: 1}]
    });
    wrapper.instance().handleUpVote();

    expect(spy).toHaveBeenCalled();
  });

  it('updates state when getNewCard is called', () => {
    const wrapper = shallow(<Sort />, { disableLifecycleMethods: true });
    wrapper.setState({
      cards: [{connection: 1}, {connection: 2}]
    });
    wrapper.instance().getNewCard();
    const results = wrapper.instance().state.cards.length;
    expect(results).toEqual(1);
  });
});