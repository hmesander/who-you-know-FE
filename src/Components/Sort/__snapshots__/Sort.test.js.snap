// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SORT TESTS should match the snapshot if there is a user id 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sort
    userId={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Card
        cardsToSort={Array []}
        handlePlay={undefined}
        handleVote={[Function]}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "cardsToSort": Array [],
        "handlePlay": undefined,
        "handleVote": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Card
          cardsToSort={Array []}
          handlePlay={undefined}
          handleVote={[Function]}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "cardsToSort": Array [],
          "handlePlay": undefined,
          "handleVote": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;

exports[`SORT TESTS should match the snapshot if there is no user id 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sort />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h6
        className="login-error-text"
>
        You must be logged in to Sort!
</h6>,
      "className": "login-error-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "You must be logged in to Sort!",
        "className": "login-error-text",
      },
      "ref": null,
      "rendered": "You must be logged in to Sort!",
      "type": "h6",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h6
          className="login-error-text"
>
          You must be logged in to Sort!
</h6>,
        "className": "login-error-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "You must be logged in to Sort!",
          "className": "login-error-text",
        },
        "ref": null,
        "rendered": "You must be logged in to Sort!",
        "type": "h6",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
